pipeline/lib/pipeline.ml

```diff
@@ -181,13 +181,29 @@ let github_repositories repo =
         repository ~commit ~github_head:head ~commit_message:message
       in
       match key with
-      (* Skip all branches other than the default branch, and check PRs *)
-      | `Ref branch when branch = default_branch ->
-          repository ~branch:default_branch_name ~labels:[] () :: lst
-      | `PR pr ->
-          repository ~title:pr.title ~pull_number:pr.id ~labels:pr.labels ()
-          :: lst
-      | _ -> lst)
+      | `Ref branch -> (
+          let branch = Util.get_branch_name branch in
+          let repository = repository ~branch ~labels:[] () in
+          match
+            List.filter
+              (Config.must_benchmark_branch ~default_branch:default_branch_name
+                 ~branch)
+              (Config.find config repository)
+          with
+          | [] -> lst
+          | _ -> repository :: lst)
```

Comment:
Not quite sure about this bit. Why drop the result of the filter instead of adding it to `lst`?
Same question for the `PR` case down

Reply:
The result of the filter is a list of config objects, where the current branch (or PR in the case below) returned by the API is configured to be benchmarked. In the case, where a whitelisted configuration exists for the current branch, we add the repository object to the list of repository objects for benchmarking.

Reply:
Ah I see. In that case I think I'd prefer something along the lines of
```ocaml
if List.exists
  (Config.must_benchmark_branch ~default_branch:default_branch_name
    ~branch)
  (Config.find config repository)
then repository :: lst
else lst
```
To make it clear that the only info we want is "is there at least one element in the list that matches the filter?"